zerver/lib/actions.py

```diff
@@ -2353,7 +2353,11 @@ def create_streams_with_welcome_messages(realm, stream_dict):
         )
 
         if created:
-            message = prep_stream_welcome_message(stream)
+            default_welcome_message = options.get('welcome_message')
+            if default_welcome_message is None:
+                message = prep_stream_welcome_message(stream)
+            else:
+                message = default_welcome_message
```

Comment:
I would do this by adding an optional topic and message argument to prep_stream_welcome_message. That way you get the benefit of all the error checking, logging, or whatever else might be happening (now or in the future) in internal_prep_stream_message.